### DISCUSSION TITLE
How to Dispatch action which depends on Http method

### DISCUSSION THREAD

**Discussion by mahmudabir** [ngxs/store#1901] [Q&A] [answered]
Which one is a good practice?
1. First `HTTP call` then `Dispatch action` in it's subscription?
2. First `Dispatch action` and then `HTTP call` in it's pipe method?

For example, what is the process of user login using state management? Here, I have to make an HTTP call & then I have to do something depending on the response I get from the HTTP response.

Which one is good practice & why?

Also, please share `example` .
Thanks in advance.

**Comment by kuncevic** [comment, marked as answer]
Usually you do an API call from the action handler within a state, then depends on what API returns, you are patch the state accordingly. Have a separate service that handle the calls. Check this SO answer for more details https://stackoverflow.com/questions/54557628/where-to-make-api-call-and-how-to-structure-actions
